src/main/java/org/dependencytrack/resources/v1/TeamResource.java

```diff
@@ -220,6 +225,33 @@ public Response deleteTeam(Team jsonTeam) {
         }
     }
 
+    @GET
+    @Path("/visible")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Returns a list of Teams what are visible", description = "Requires permission <strong>PORTFOLIO_MANAGEMENT</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "The Visible Teams", content = @Content(schema = @Schema(implementation = VisibleTeams.class))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "Teams could not be found")
+    })
+    public Response availableTeams() {
+        try (QueryManager qm = new QueryManager()) {
+            Principal user = getPrincipal();
+            List<Team> userTeams = new ArrayList<Team>();
+            if (user instanceof final UserPrincipal userPrincipal) {
+                userTeams = userPrincipal.getTeams();
+            } else if (user instanceof final ApiKey apiKey) {
+                userTeams = apiKey.getTeams();
+            }
```

Comment:
@ nscuro Would it suffice to only do that for the UserPrincipal? As API Keys are only for teams, so there is only one team to return, and mostly this and the /project calls are done by users, aren't their?

Reply:
The API key's team could have the `ACCESS_MANAGEMENT` permission, in which case we'd need to return all teams here.

Also more generally, for consistency, let's keep handling of API keys in here. Even if it will just return one team most of the time.